sqlglot/dialects/spark.py

```diff
@@ -111,16 +113,22 @@ class Parser(Hive.Parser):
             "SHUFFLE_REPLICATE_NL": lambda self: self._parse_join_hint("SHUFFLE_REPLICATE_NL"),
         }
 
-        def _parse_add_column(self):
+        def _parse_add_column(self) -> t.Optional[exp.Expression]:
             return self._match_text_seq("ADD", "COLUMNS") and self._parse_schema()
 
-        def _parse_drop_column(self):
+        def _parse_drop_column(self) -> t.Optional[exp.Expression]:
             return self._match_text_seq("DROP", "COLUMNS") and self.expression(
                 exp.Drop,
                 this=self._parse_schema(),
                 kind="COLUMNS",
             )
 
+        def _pivot_column_names(self, pivot_columns: t.List[exp.Expression]) -> t.List[str]:
+            # Spark doesn't add a suffix to the pivot columns when there's a single aggregation
+            if len(pivot_columns) == 1:
+                return [""]
+            return [agg.alias or agg.sql(dialect="spark").lower() for agg in pivot_columns]
```

Comment:
why is it lower?

Reply:
That's the name we get back when running spark. Observe:

```sql
spark-sql> WITH person(id, name, age, class, address) AS (SELECT * FROM VALUES (100, 'John', 30, 1, 'Street 1'), (200, 'Mary', NULL, 1, 'Street 2'), (300, 'Mike', 80, 3, 'Street 3'), (400, 'Dan', 50, 4, 'Street 4')) SELECT * FROM person PIVOT (SUM(age), AVG(class) AS c FOR name IN ('John', 'Mike' AS mike));
id	address	John_sum(age)	John_c	mike_sum(age)	mike_c
200	Street 2	NULL	NULL	NULL	NULL
100	Street 1	30	1.0	NULL	NULL
300	Street 3	NULL	NULL	80	3.0
400	Street 4	NULL	NULL	NULL	NULL
Time taken: 0.653 seconds, Fetched 4 row(s)
```

Reply:
what if it's quoted?

Reply:
Good point, it doesn't work well with quoted names that might contain uppercase letters. The issue is that function names are converted to lowercase in spark and so we need to mimic that somehow, but also respect the quoting / casing of quoted identifiers used as arguments to these aggregations.

So if we had `... PIVOT (SUM(`AgE`), .....)`,, we'd need to convert this to `John_sum(AgE)` (quoted).

Reply:
Do you think I can just do `.sql(..., normalize_func="lower")` for this?

Reply:
normalize_func is different than the identifiers right?

Reply:
It simply controls whether we're going to output ``` SUM(`FoO`) ``` as is or convert `SUM` to lowercase.

Reply:
Changed it

Reply:
Ah damn there's still an issue because it parses it in the following identifier (will be double-quoted when doing .sql)
```
(IDENTIFIER this: John_sum(`FoO`), quoted: True)
```

Reply:
Fyi spark does:
```sql
spark-sql> WITH person(id, name, `AgE`, class, address) AS (SELECT * FROM VALUES (100, 'John', 30, 1, 'Street 1'), (200, 'Mary', NULL, 1, 'Street 2'), (300, 'Mike', 80, 3, 'Street 3'), (400, 'Dan', 50, 4, 'Street 4')) SELECT * FROM person PIVOT (SUM(`AgE`), AVG(class) AS c FOR name IN ('John', 'Mike' AS mike));
id	address	John_sum(AgE)	John_c	mike_sum(AgE)	mike_c
200	Street 2	NULL	NULL	NULL	NULL
100	Street 1	30	1.0	NULL	NULL
300	Street 3	NULL	NULL	80	3.0
400	Street 4	NULL	NULL	NULL	NULL
Time taken: 1.145 seconds, Fetched 4 row(s)
```